packages/marshal/src/passStyleOf.js

```diff
@@ -3,598 +3,197 @@
+const { passStyleOf, HelperTable } = makePassStyleOfKit([
+  CopyArrayHelper,
+  CopyRecordHelper,
+  ErrorHelper,
+  RemotableHelper,
+]);
 
-/**
- * @param { Passable } val
- * @param { Set<Passable> } inProgress
- * @returns { PassStyle }
- */
-const passStyleOfRecur = (val, inProgress) => {
-  const isObject = Object(val) === val;
-  if (isObject) {
-    if (passStyleOfCache.has(val)) {
-      // @ts-ignore
-      return passStyleOfCache.get(val);
-    }
-    assert(!inProgress.has(val), X`Pass-by-copy data cannot be cyclic ${val}`);
-    inProgress.add(val);
-  }
-  const passStyle = passStyleOfInternal(val, inProgress);
-  if (isObject) {
-    passStyleOfCache.set(val, passStyle);
-    inProgress.delete(val);
+export { passStyleOf };
+
+export const everyPassableChild = (passable, fn) => {
+  const passStyle = passStyleOf(passable);
+  const helper = HelperTable[passStyle];
+  if (helper) {
+    // everyPassable guards .every so that each helper only gets a
+    // genuine passable of its own flavor.
+    return helper.every(passable, fn);
   }
-  return passStyle;
+  return true;
 };
-
-/**
- * objects can only be passed in one of two/three forms:
- * 1: pass-by-remote: all properties (own and inherited) are methods,
- *    the object itself is of type object, not function
- * 2: pass-by-copy: all string-named own properties are data, not methods
- *    the object must inherit from objectPrototype or null
- * 3: the empty object is pass-by-remote, for identity comparison
- *
- * all objects must be frozen
- *
- * anything else will throw an error if you try to serialize it
- * with these restrictions, our remote call/copy protocols expose all useful
- * behavior of these objects: pass-by-remote objects have no other data (so
- * there's nothing else to copy), and pass-by-copy objects have no other
- * behavior (so there's nothing else to invoke)
- *
- * How would val be passed?  For primitive values, the answer is
- *   * 'null' for null
- *   * throwing an error for a symbol, whether registered or not.
- *   * that value's typeof string for all other primitive values
- * For frozen objects, the possible answers
- *   * 'copyRecord' for non-empty records with only data properties
- *   * 'copyArray' for arrays with only data properties
- *   * 'copyError' for instances of Error with only data properties
- *   * 'remotable' for non-array objects with only method properties
- *   * 'promise' for genuine promises only
- *   * throwing an error on anything else, including thenables.
- * We export passStyleOf so other algorithms can use this module's
- * classification.
- *
- * @param {Passable} val
- * @returns {PassStyle}
- */
-export const passStyleOf = val =>
-  // Even when a WeakSet is correct, when the set has a shorter lifetime
-  // than its keys, we prefer a Set due to expected implementation
-  // tradeoffs.
-  passStyleOfRecur(val, new Set());
-harden(passStyleOf);
+harden({ everyPassable: everyPassableChild });
```

Comment:
This looks like leftovers from a code refactor.

```suggestion
harden(everyPassableChild);
```

Reply:
oops. correct.